app/services/trainees/create_from_hesa.rb

```diff
@@ -171,10 +171,11 @@ def disabilities
     end
 
     def enqueue_background_jobs!
-      if FeatureService.enabled?(:integrate_with_dqt)
-        Trainees::Update.call(trainee:) if trainee.trn.present?
-        Dqt::RegisterForTrnJob.perform_later(trainee) if request_for_trn?
-      end
+      return unless FeatureService.enabled?(:integrate_with_dqt)
+
+      Trainees::Update.call(trainee:) if trainee.trn.present?
+      Dqt::RegisterForTrnJob.perform_later(trainee) if request_for_trn?
+      Dqt::WithdrawTraineeJob.perform_later(trainee) if trainee_state == :withdrawn
```

Comment:
i feel like we'll get some race conditions happening here but not sure what the answer is..

1. If the trainee comes across as withdrawn without a TRN, then we request a TRN and withdraw them straight after. What would DQT do in this scenario? i presume the withdraw job will 404 if it gets through first. Will it succeed if it arrives immediately after?

2. If the trainee already has a TRN and is updated, then we kick off a DQT update job and withdrawn them straight after. This may mean that the update job fails with the "no incomplete ITT record" error. This already occasionally happens when we award trainees.

Reply:
You're right @AbigailMcP - good spot.

I've updated the code so that in scenario 1, if the trainee comes across as withdrawn without a TRN, the TRN is requested first and `Dqt::WithdrawTraineeJob` will wait until the TRN has been issued. In scenario 2, the update will happen first (as it should) and `Dqt::WithdrawTraineeJob` will only kick in an hour later.

`Dqt::WithdrawTraineeJob` is configured to run 1 hour later in both scenarios allowing time for other jobs to be completed first. This is the simplest solution I've come up with while trying to add more complexity.

Reply:
cc @claire-hughez this is what we were talking about the other day